lib/network_interface/worker.ex

```diff
@@ -13,45 +13,105 @@
 # limitations under the License.
 
 defmodule Nerves.NetworkInterface.Worker do
+  @moduledoc """
+  Working for NetworkInterface. 
+  See `Nerves.NetworkInterface` for more details.
+  """
+
   use GenServer
   require Logger
 
-  @moduledoc false
+  defstruct port: nil
+
+  @typedoc "State of the GenServer"
+  @type state :: %__MODULE__{port: port}
```

Comment:
If the struct type requires that a port to be set I think it would be good to add `@enforce_keys [:port]`, or allow the type to include `nil` on the port.